web-admin/src/features/bookmarks/BookmarksFormDialog.svelte

```diff
@@ -38,33 +55,90 @@
   export let bookmark: BookmarkEntry | null = null;
   export let defaultUrlParams: URLSearchParams | undefined = undefined;
   export let showFiltersOnly: boolean = true;
-  export let dashboardState: {
-    metricsViewNames: string[];
-    filtersState: FiltersState;
-    timeControlState: TimeControlState;
-  };
+  export let metricsViewNames: string[];
   export let onClose = () => {};
 
+  let filterState = baseFilterState;
+
   $: ({ name: resourceName, kind: resourceKind } = resource);
-  $: ({ metricsViewNames, filtersState, timeControlState } = dashboardState);
 
   $: ({ url } = $page);
   $: curUrlParams = url.searchParams;
 
-  $: start = timeControlState?.selectedTimeRange?.start?.toISOString() ?? "";
-  $: end = timeControlState?.selectedTimeRange?.end?.toISOString() ?? "";
-  $: timeRange = <V1TimeRange>{
-    isoDuration: timeControlState.selectedTimeRange?.name,
-    start,
-    end,
-  };
+  $: bookmarkUrl = bookmark?.url || $page.url.searchParams.toString();
+
+  $: processUrl(bookmarkUrl)
+    .then((state) => {
+      filterState = state;
+    })
+    .catch(console.error);
+
+  async function processUrl(searchParams: string) {
```

Comment:
Would be good to move this to some common place and reuse as much of it as possible in both explore and canvas. But out of scope for this PR